examples/plot_unified_validation_scheme.py

```diff
@@ -0,0 +1,344 @@
+# ``test_size`` parameter. Here we use a 80-20 train-val split.
+#
+# .. note::
+#    The parameter ``shuffle`` is set to ``False``. For time-series
+#    data this should always be the case as shuffling might take
+#    advantage of correlated samples, which would make the validation
+#    performance less significant.
+#
+
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import GridSearchCV, KFold, train_test_split
+from skorch.helper import SliceDataset
+
+X_train = SliceDataset(train_set, idx=0)
+y_train = np.array([y for y in SliceDataset(train_set, idx=1)])
+train_val_split = [tuple(train_test_split(X_train.indices_, test_size=0.2, shuffle=False))]
+
+######################################################################
+# Define the ``fit_params`` and the ``paramter_grid`` i.e. list all
+# Hyperparamters you want to include in your search.
+# Afterwards define a search strategy. As a simple example we use
+# grid search, but you can use `any strategy you want
+# <https://scikit-learn.org/stable/modules/classes.html#module-sklearn.model_selection>`__).
+#
+
+fit_params = {'epochs': n_epochs}
+param_grid = {
+    'optimizer__lr': [0.00625, 0.000625],
+}
+search = GridSearchCV(
+    estimator=clf,
+    param_grid=param_grid,
+    cv=train_val_split,
+    return_train_score=True,
+    scoring='accuracy',
+    refit=True,
+    verbose=1,
+    error_score='raise'
+)
+
+search.fit(X_train, y_train, **fit_params)
+search_results = pd.DataFrame(search.cv_results_)
+
+best_run = search_results[search_results['rank_test_score'] == 1].squeeze()
+print(f"Best hyperparameters were {best_run['params']} which gave a validation "
+      f"accuracy of {best_run['mean_test_score'] * 100:.2f}% (training "
+      f"accuracy of {best_run['mean_train_score'] * 100:.2f}%).")
+
+X_test = SliceDataset(test_set, idx=0)
+y_test = SliceDataset(test_set, idx=1)
+test_acc = search.score(X_test, y_test)
+print(f"Test accuracy: {test_acc * 100:.2f}%.")
+
+######################################################################
+# Option 3: Hyperparameter Search via Cross Validation
+# ----------------------------------------------------
+#
+
+######################################################################
+# As mentioned above, using only one validation split might not be
+# sufficient, as there might be a shift in the data distribution.
+# To compensate this, one can run a k-fold Cross Validation, where
+# every sample of the training set is in the validation set once.
+# By averaging over the k validation scores afterwards, you get a
+# very reliable estimate of how the model would perform on unseen
+# data (test set).
+# The implementation is straight forward as we only have to change one
+# line of code compared to Option 2. The ``KFold`` class will split
+# the ``train_set`` into k folds. We will use 5 folds to get a 80-20
+# train-val split.
+#
+
+train_val_split = KFold(n_splits=5, shuffle=False)
```

Comment:
maybe you can also show how to evaluate model on cross-validation with `cross_val_score` - should be one line somewhere, just to show that it can be also used as performance evaluation method and not only hyperparams search

Reply:
Good point - I will add that